packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -139,6 +139,31 @@ function Thread( {
 			id={ thread.id }
 			spacing="3"
 			onClick={ () => handleCommentSelect( thread ) }
+			onKeyDown={ ( event ) => {
+				if ( event.key === 'Enter' || event.key === ' ' ) {
+					event.preventDefault();
+					handleCommentSelect( thread.id );
+				}
+			} }
+			role="button"
+			tabIndex={ 0 }
+			aria-label={ sprintf(
+				// translators: %1$s: comment identifier, %2$s: author name, %3$s: reply count, %4$s: status, %5$s: date.
+				_x(
+					'Comments dialog. Open comment %1$s. Author %2$s. %3$s replies. Status: %4$s. Posted %5$s. Click to focus.',
+					'Comment accessibility label with full context'
+				),
+				thread.id,
+				thread?.author_name || 'Unknown',
+				thread?.replies?.length || 0,
+				thread?.status === 'approved' ? 'resolved' : 'active',
+				thread?.date
+					? new Date( thread.date ).toLocaleDateString()
+					: 'unknown date'
+			) }
+			{ ...( thread?.replies?.length > 0 && {
+				'aria-expanded': isFocused,
+			} ) }
```

Comment:
I don't think these changes are necessary; the aria-label on the thread container itself could be discussed in https://github.com/WordPress/gutenberg/pull/71883.